root/scripts/utils.js

```diff
@@ -312,9 +321,33 @@ export async function search(searchQuery, tags) {
       let numMatchingCharacters = 0;
       let searchScore = 0;
       const { title } = recipe;
+      const { ingredients } = recipe;
       if (title) {
         for (let j = tokenizedQuery.length - 1; j >= 0; j -= 1) {
           if (title.toLowerCase().includes(tokenizedQuery[j])) {
+            const ingreds = JSON.stringify(ingredients);
+            // if ingredients are being filtered, check
+            let containsEveryIngredient = true;
+            splitIngredientsIncluded.every((currIngredient) => {
+              if (!ingreds.includes(currIngredient)) {
+                containsEveryIngredient = false;
+                return false;
+              }
+              return true;
+            });
+
+            let excludesAnyIngredient = true;
+            splitIngredientsExcluded.every((currentIngredient) => {
+              if (ingreds.includes(currentIngredient)) {
+                excludesAnyIngredient = false;
+                return false;
+              }
+              return true;
+            });
+
+            if (!containsEveryIngredient || !excludesAnyIngredient) {
```

Comment:
NICE! This is probably the best way to accomplish ingredient search, you're right in that we probably don't want to generate tokenized scores for ingredients like we do titles (because it's a hard inclusion exclusion scenario).